medusa/show/recommendations/recommended.py

```diff
@@ -131,3 +138,89 @@ def check_if_anime(self, anidb, tvdbid):
     def __str__(self):
         """Return a string repr of the recommended list."""
         return 'Recommended show {0} from recommended list: {1}'.format(self.title, self.recommender)
+
+
+@recommended_series_cache.cache_on_arguments()
+def cached_tvdb_to_aid(tvdb_id):
+    """
+    Try to match an anidb id with a tvdb id.
+
+    Use dogpile cache to return a cached id if available.
+    """
+    return anidb_api.tvdb_id_to_aid(tvdbid=tvdb_id)
+
+
+@recommended_series_cache.cache_on_arguments()
+def cached_aid_to_tvdb(aid):
+    """
+    Try to match a tvdb id with an anidb id.
+
+    Use dogpile cache to return a cached id if available.
+    """
+    return anidb_api.aid_to_tvdb_id(aid=aid)
+
+
+@recommended_series_cache.cache_on_arguments()
+def cached_get_imdb_series_details(imdb_id):
+    """
+    Request the series details from the imdbpie api.
+
+    Use dogpile cache to return a cached id if available.
+    """
+    return imdb_api.get_title(imdb_id)
+
+
+def create_key_from_series(namespace, fn, **kw):
+    """Generate a key limiting the amount of dictionaries keys that are allowed to be used."""
+
+    def generate_key(*arg):
+        """
+        Generate the key.
+        The dogpiled items should have the following format:
+            `dict(keys={'key1': 'value_for_key1', 'key2': 'value_for_key2}, item=to_be_cached_item)`.
+        Following this standard we can cache every object, using this key_generator.
+        """
+        keys = arg[1]['keys']
+        return b'{arguments}'.format(
+            arguments=b'_'.join(binary_type(v) for v in keys)
+        )
+
+    return generate_key
+
+
+def update_recommended_series_cache_index(indexer, new_index):
```

Comment:
@ratoaq2 i've created this function to store an index of all the keys that i've added.
do you know if there is already a native dogpile solution for this? I want to do something like return all shows with this namespace for example.